src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ApplicationAssembliesProvider.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    internal class ApplicationAssembliesProvider
```

Comment:
Why are we killing DefaultAssemblyPartDiscoveryProvider and making this new type internal? It seems a bit of moving the cheese around, if we can just do this on the existing type we should avoid changing it.

Reply:
We're generally replacing `pub-internal` types with regular `internal` when we get the opportunity. That said, the type no longer discovers parts, it discovers assemblies, the naming change reflects that.

Reply:
It might be that this needs to be public instead, but fair enough. If someone complains we can react